src/network/address.rs

```diff
@@ -69,8 +69,11 @@ impl Address {
 }
 
 fn addr_to_be(addr: [u16; 8]) -> [u16; 8] {
-    [addr[0].to_be(), addr[1].to_be(), addr[2].to_be(), addr[3].to_be(),
-     addr[4].to_be(), addr[5].to_be(), addr[6].to_be(), addr[7].to_be()]
+    let mut result = addr.clone();
+    for i in 0..8 {
+        result[i] = result[i].swap_bytes();
```

Comment:
It's not obvious to me why it should be `swap_bytes` and not `to_be`, could you expand on that?

Reply:
To be completely honest I've seen successful test on little-endian failing with swapped bytes on big-endian arch, so I checked if failing tests had test vector shared with bitcoin core, and I've see it's the case for `[de]serialize_addrv2_test`, so I swapped bytes also on big-endian (`to_be()` is a noop on big-endian) to make tests happy ( I need to rename the fn to something like`addr_to_swapped`...)

Reply:
@elichai because `consensus_encode` always encodes in LE, and we want to instead always encode in BE.

I agree it's confusing but this is probably the most sensible thing to do..

Reply:
Maybe we should instead change the address type to something other than 8 u16s? u16s are pretty strange for address representation IMO.

Reply:
I think it makes sense to change the address type, but should be done in another PR since this is a bugfix